### ISSUE TITLE
`write` allows for invalid shell variable creation

### ISSUE THREAD

**Issue by jmhodges-color** [segmentio/chamber#366]
This is with chamber v2.11.0 on macOS 13.1

I think chamber should be more strict about what it accepts in names at `write` time. For instance, accepting dashes (`-`) in variable names due to two related, but interestingly different bugs described below in `env` and `export -f dotenv`.

Accepting periods (`.`) also causes invalid env var names to be created in both `env` and `export -f dotenv`, but those are not described here.

Both suppose you have written a secret to chamber with a dash in it:

```
chamber write mysvc foo-bar 1
```

### env bug
if run `chamber env mysvc`, you'll get the invalid variable `FOO-BAR` in the output. The dash in it is not supported in shell variables in both `bash` and `zsh`.

```
$ chamber env mysvc
export FOO-BAR=1

$ source <(chamber env mysvc)
/dev/fd/11:export:1: not valid in this context: FOO-BAR

```

### export dotenv bug

If you run a second write with an underscore version of the parameter, then run export with the `dotenv` format, you'll get duplicate env vars with no warning. (Interestingly, the export dotenv code path does normalize the dash to an underscore in this case)

```
$ chamber write mysvc foo_bar 2
$ chamber export -f dotenv mysvc
FOO_BAR=2
FOO_BAR=1
$ 
```

### Conclusion
We're planning on telling folks to not use dashes from variable names, but would like to have some support from `chamber` to prevent these kinds of problems at time of `write`.

**Comment by alecjacobs5401** [comment]
Thanks for reporting! 

>if run `chamber env mysvc`, you'll get the invalid variable `FOO-BAR` in the output. The dash in it is not supported in shell variables in both `bash` and `zsh`.

This feels like the core bug here and I believe there are some other issues that relate. This might warrant a separate issue

> If you run a second write with an underscore version of the parameter, then run export with the `dotenv` format, you'll get duplicate env vars with no warning.

This could be related to lack of validation testing on duplicate keys where case insensitivity/normalization is not respected, which leads to duplicate keys that effectively resolve to the same environment variable (related https://github.com/segmentio/chamber/pull/162, https://github.com/segmentio/chamber/issues/160)

Due to the breaking nature of the latter problem, it would likely need to be part of a 3.0 release.

I'll bring this up with my team to prioritize looking into

**Comment by mckern** [comment]
@alecjacobs5401 I have taken a look at this, and I think that this is worth prioritizing around "as soon as we can get to it" and breaking semver over; collectively, we never should have let `env` out the door without validating that printed keys were in a valid format. I know that this predates both of our time at Segment, but we should make this right or remove the command -- it's borderline unusable the way it works now. I'm really surprised there's no PRs related to this and I wonder how many folks use the `env` subcommand (or how many tried, saw it was broken, and just gave up).

**Comment by mckern** [comment]
@jmhodges-color I just landed PR #385, which fixes this on the *output* but not the creation of new secrets. I believe that this is the less disruptive change overall right now but may seek to impose similar rules on creation in the future re: a hard `v3` break from the `v2` code).
